src/controller/c/robot.c

```diff
@@ -1085,8 +1092,10 @@ int wb_robot_init() {  // API initialization
   robot.has_html_robot_window = false;
   robot.wwi_message_to_send = NULL;
   robot.wwi_message_to_send_size = 0;
-  robot.wwi_message_received = NULL;
-  robot.wwi_message_received_size = 0;
+  robot.wwi_reception_buffer = NULL;
+  robot.wwi_received_messages_size = 0;
+  robot.wwi_reception_buffer_size = 0;
+  robot.wwi_reset_read_head = true;
```

Comment:
That looks good to me. I would just name the new variable this way:
```suggestion
  robot.wwi_reset_reading_head = true;
```